### PR TITLE
:sparkles: Add tracdelight AMP ad integration

### PR DESCRIPTION
**Pull request by thekorn**
The pull request contains implementation for the [tracdelight](https://www.tracdelight.com) ad-widgets for AMP Ads.

### CODE REVIEW

**Comment by thekorn** [comment]
My company just signed the CLA and I'm an authorized contributor

File: ads/tracdelight.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        type="tracdelight"
+        data-mode="inline"
+        data-widget_id="[widget-id]"
+        data-access_key="[access-key]">
+</amp-ad>
+```
+
+## Configuration
+
+For semantics of configuration, please see ad network documentation.
```
Comment (zhouyx) [inline ads/tracdelight.md]: Please link to your documentation.

File: ads/tracdelight.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Supported parameters:
+
+- data-widget_id — the id of the widget
```
Comment (zhouyx) [inline ads/tracdelight.md]: Please indicate whether a param is required or optional.

File: ads/_config.js [hunk 3]
```diff
@@ -845,6 +845,10 @@ export const adConfig = {
     ],
   },
 
+  tracdelight: {
+    prefetch: 'https://scripts.tracdelight.com/amp.js',
```
Comment (zhouyx) [inline ads/_config.js]: There is a `renderStart` API that helps to deliver better user experiences, and we highly recommend you implement it. Please refer to the doc [here](https://github.com/ampproject/amphtml/blob/master/ads/README.md#available-apis) and let me know if there's any questions.

File: ads/tracdelight.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+import {validateData, writeScript} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function tracdelight(global, data) {
+  validateData(data, [], ['widget_id', 'access_key']);
```
Comment (zhouyx) [inline ads/tracdelight.js]: I am assuming both `data-widget_id` and `data-access_key` are optional. Same with `data-mode`?

File: examples/tracdelight.amp.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
+<!doctype html>
```
Comment (zhouyx) [inline examples/tracdelight.amp.html]: No need for individual example page. Please revert.

File: ads/tracdelight.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+<!---
```
Comment (zhouyx) [inline ads/tracdelight.md]: Please link to your readme from the amp-ad.md.

**Comment by nainar** [comment]
@thekorn is this something you are working on actively? To reduce the stale PRs I will close this for now since there hasn't been action on this in 2 months. Please reopen this if you intend to work on this.

**Comment by thekorn** [comment]
@nainar - yes, we are still working on this. We are just figuring out how to address the most recent review comments

**Pull request closed without merging**